### PR TITLE
Refactor/drivers config

### PR DESCRIPTION
**Pull request by collindutter**
- [x] I have read and agree to the contributing guidelines for [submitting new pull requests](https://github.com/griptape-ai/griptape?tab=readme-ov-file#submitting-pull-requests).

## Describe your changes
- Rename `DriverConfig` to `DriversConfig` to indicate that it's _multiple_ Drivers in a configuration.
- Rename `griptape.config` to `griptape.configs` for consistency with other packages (`gritpape.drivers`, `griptape.artifacts`, etc).
- Rename `griptape.configs.config._Config` to `griptape.configs.defaults_config._DefaultsConfig`.
- Rename `griptape.configs.config` to `griptape.configs.defaults_config.Defaults`.
- Rename `griptape.events.event_bus` to `griptape.events.EventBus`.
- Add `griptape.mixins.SingletonMixin`.
## Issue ticket number and link
Offline discussion

----
📚 Documentation preview 📚: https://griptape--1086.org.readthedocs.build//1086/

### CODE REVIEW

File: docs/griptape-framework/structures/drivers-configs.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+--8<-- "docs/griptape-framework/structures/src/drivers_config_7.py"
+```
+
+### Logging Config
+
+Griptape provides a predefined [LoggingConfig](../../reference/griptape/configs/logging/logging_config.md)'s for easily customizing the logging events that the framework emits. In order to customize the logger, the logger can be fetched by using the `config.logging.logger_name`.
+
+```python
+--8<-- "docs/griptape-framework/structures/src/logging_config.py"
+```
```
Comment (vachillo) [inline docs/griptape-framework/structures/drivers-configs.md:83]: move this to its own docs page

**Review by vachillo — COMMENTED** [review]
should the `config` object be called something else? `global_config` or `default_config` or something?

File: docs/griptape-framework/structures/drivers-configs.md (hunk 1 above)
Comment (collindutter) [reply docs/griptape-framework/structures/drivers-configs.md]: Renamed back to a general `configs.md`

**Review by vachillo — APPROVED** [review]
the only constant is change

**Pull request merged**